driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -665,7 +754,7 @@ public void shutdown() {
     }
 
     private static final class Flusher implements Runnable {
-        final EventLoop eventLoop;
+        final WeakReference<EventLoop> eventLoopRef;
```

Comment:
Yes, the only reason Flusher keeps a reference to the event loop is to reschedule itself, which makes the whole thing kind of weird (if `run()` is executing, the event loop has obviously not been GC'ed). But I agree that using `WeakReference` here feels slightly better.

On a side note, since this code was copied from Cassandra, couldn't it cause a leak there as well? Event loops also become stale as clients come and go. It's probably less severe, but worth reporting.